src/main/java/com/bandwidth/iris/sdk/IrisClient.java

```diff
@@ -1,186 +0,0 @@
-        if (parsedResponse instanceof BaseResponse){
-            BaseResponse baseResponse = (BaseResponse) parsedResponse;
-            if(baseResponse.getResponseStatus() != null){
-                throw new IrisClientException(baseResponse.getResponseStatus().getErrorCode(),
-                        baseResponse.getResponseStatus().getDescription());
-            }
-        }
-        return parsedResponse;
-    }
-
-    public <T> T get(String uri, Class<T> returnType) throws Exception {
-        IrisResponse response = get(uri);
-        return processResponse(response.getResponseBody(), returnType);
-    }
-
-    public IrisResponse get(String uri) throws Exception {
-        HttpGet get = new HttpGet(uri);
-        return executeRequest(get);
-    }
-
-    public <T> T post(String uri, BaseModel data, Class<T> returnType) throws Exception {
-        IrisResponse response = post(uri, data);
-        return processResponse(response.getResponseBody(), returnType);
-    }
-    public IrisResponse post(String uri, BaseModel data) throws Exception {
-        HttpPost post = new HttpPost(uri);
-        StringEntity postBody = new StringEntity(XmlUtils.toXml(data));
-        post.addHeader("Content-Type", "application/xml");
-        post.setEntity(postBody);
-        return executeRequest(post);
-    }
-
-    public IrisResponse delete(String uri) throws Exception {
-        HttpDelete delete = new HttpDelete(uri);
-        return executeRequest(delete);
-    }
-
-    public <T> T put(String uri, BaseModel data, Class<T> returnType) throws Exception {
-        IrisResponse response = put(uri, data);
-        return processResponse(response.getResponseBody(), returnType);
-    }
-    public IrisResponse put(String uri, BaseModel data) throws Exception {
-        HttpPut put = new HttpPut(uri);
-        return executeRequest(put);
-    }
-
-    public void postFile(String uri, File file, String contentType) throws Exception {
-        HttpPost post = new HttpPost(uri);
-        post.setEntity(contentType == null ? new FileEntity(file) : new FileEntity(file, ContentType.parse(contentType)));
-        executeRequest(post);
-    }
-
-    public void putFile(String uri, File file, String contentType) throws Exception {
-        HttpPut put = new HttpPut(uri);
-        put.setEntity(contentType == null ? new FileEntity(file) : new FileEntity(file, ContentType.parse(contentType)));
-        executeRequest(put);
-    }
-
-    public String buildUserModelUri(String uriSuffix, Map<String, Object> query) throws URISyntaxException {
-        URIBuilder builder = new URIBuilder(this.uri);
-        builder.setPath(baseUserUrl + uriSuffix);
-        if (query != null) {
-            for (String key : query.keySet()) {
-                builder.addParameter(key, query.get(key).toString());
-            }
-        }
-        return builder.build().toString();
-    }
-
-    public String buildUserModelUri(String[] tokens) throws URISyntaxException {
```

Comment:
if you want to be really nice to you users make this a var args, then users won't need to first create an array
`public String buildUserModelUri(String... tokens)`
and in use it would just be:
`buildUserModel(firstToken, secondToken, moreTokens)`